internal/kafka/internal/api/public/api/openapi.yaml

```diff
@@ -283,6 +283,89 @@ paths:
           description: Unexpected error occurred
       security:
       - Bearer: []
+  /api/kafkas_mgmt/v1/kafkas/{id}/promote:
```

Comment:
This brings us back to a discussion we had a few days ago. We are not creating a resource with this endpoint: we are updating the `kafka` resource.

We should use the `update Kafka endpoint from a REST POV (IMO).
That, however, led to the issues you already exposed in the PR description.

On the other hand, since we don't have a `promotion` resource, I also believe that using a `promotion` endpoint is not the best option.

Following those reasonings, I believe that the `promote` endpoint is the best option.
I'd be interested to know what others think about it: @machi1990 @JameelB

Reply:
Agree with @ziccardi, I think it makes more sense to have `promote` for this endpoint. the cluster service of ocm does have some verbs in their api as well (i.e. /resume and /hibernate)

Reply:
I think going for `promote` also makes sense. Although not "pure" from a REST point of view I think it denotes better that it is an action more than a resource in this case and it also leaves space to become a resource in the future if needed by adding the endpoint name `promotion` and deprecating the `promote` one.

I will keep the `promote` name for the endpoint